### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Resolves: *(direct link to the issue)*

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/barrier/commit/399aacd84dd96da4fc59d2f48571436e4758e326#commitcomment-72989896

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/barrier/commit/870431b06d239879af6216c1f6fbcb85af4f4f0e

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I'll be annotating this PR, but, it's a long PR, so please bare with me.
Note: it's quite likely that I shouldn't change certain files or certain spellings. I'm happy to drop entire files, any sets of changes, or add an upstream to my todo list. I'm also happy to split this PR into pieces -- just give me suggestions about how you'd like it split.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [ ] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [ ] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/framework/mpe/soundid.h
Comment (jsoref) [inline src/framework/mpe/soundid.h:88]: I think this change is right.

File: test/lines2.msc
Comment (jsoref) [inline test/lines2.msc:280]: Curious about this

File: src/engraving/libmscore/layoutbreak.cpp
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/layoutbreak.cpp:115]: Probably needs a compatibility check, like 
} else if (tag == "firstSystemIndentation" || tag == "firstSystemIdentation") {

File: fonts/smufl/glyphnames.json
Comment (Jojo-Schmitz) [reply fonts/smufl/glyphnames.json:1805]: This is taken straigth from SMuFL, so would need to get fixed upstream (too/first). See https://w3c.github.io/smufl/releases/1.4/tables/other-accidentals.html?msclkid=cf3e4ec6ceac11eca537594caf8d2f4e

File: fonts/petaluma/OFL-FAQ.txt
Comment (Jojo-Schmitz) [inline fonts/petaluma/OFL-FAQ.txt:368]: Taken from https://github.com/steinbergmedia/petaluma, so wouzld need to get fixed there (too/first)

File: fonts/bravura/OFL-FAQ.txt
Comment (Jojo-Schmitz) [inline fonts/bravura/OFL-FAQ.txt:368]: Taken from https://github.com/steinbergmedia/bravura, so would need to get fixed there (too/first)

File: fonts/bravura/FONTLOG.txt
Comment (Jojo-Schmitz) [reply fonts/bravura/FONTLOG.txt:172]: Taken from https://github.com/steinbergmedia/petaluma, so would need to get fixed there (too/first)

File: fonts/bravura/FONTLOG.txt
Comment (Jojo-Schmitz) [inline fonts/bravura/FONTLOG.txt:325]: Taken from https://github.com/steinbergmedia/petaluma, so wouzld need to get fixed there (too/first)

File: share/instruments/instruments.xml
Comment (Jojo-Schmitz) [reply share/instruments/instruments.xml:2875]: Stems from https://github.com/w3c/musicxml/blob/49f2fd5b757f2ca251a3f5448a11e386e5dd42a9/schema/sounds.xml#L897, is correct there, so definitly a typo here. 
However: it needs to get fix in the doc @shoogle maintains and is used to generate this file

File: share/instruments/instruments.xml
Comment (Jojo-Schmitz) [reply share/instruments/instruments.xml:5277]: @shoogle ?

File: src/engraving/libmscore/layoutbreak.cpp
Comment (jsoref) [reply src/engraving/libmscore/layoutbreak.cpp:115]: My general preference would be to fix this in a distinct PR.

It's bad to mix general "spelling fixes" w/ "API changes" even if they're clean.

That said, I'm flexible.

**Comment by Jojo-Schmitz** [comment]
One Braille unit test failed. Not sure why
```
2022-05-08T09:33:04.0190974Z 9: 09:33:04.018 | ERROR | main_thread | compat     | mscxToMscz: failed open file: "/home/runner/work/MuseScore/MuseScore/src/importexport/braille/tests/data/testGlissando_Example_16.6.1_MBC2015.mscx" 
2022-05-08T09:33:04.0192029Z 9: 09:33:04.018 | ERROR | main_thread | MTest      | readScore: cannot load file at "/home/runner/work/MuseScore/MuseScore/src/importexport/braille/tests/data/testGlissando_Example_16.6.1_MBC2015.mscx" 
2022-05-08T09:33:04.0203338Z 9: FAIL!  : TestBrailleIO::glissando() 'score' returned FALSE. ()
```

**Comment by HemantAntony** [comment]
> One Braille unit test failed. Not sure why
> 
> ```
> 2022-05-08T09:33:04.0190974Z 9: 09:33:04.018 | ERROR | main_thread | compat     | mscxToMscz: failed open file: "/home/runner/work/MuseScore/MuseScore/src/importexport/braille/tests/data/testGlissando_Example_16.6.1_MBC2015.mscx" 
> 2022-05-08T09:33:04.0192029Z 9: 09:33:04.018 | ERROR | main_thread | MTest      | readScore: cannot load file at "/home/runner/work/MuseScore/MuseScore/src/importexport/braille/tests/data/testGlissando_Example_16.6.1_MBC2015.mscx" 
> 2022-05-08T09:33:04.0203338Z 9: FAIL!  : TestBrailleIO::glissando() 'score' returned FALSE. ()
> ```

@jsoref renamed those two files

**Comment by jsoref** [comment]
I missed a rename, the latest push should resolve it.

I have tooling for renames, but I don't run it by default. Out of ~240 commits, I think ~half a dozen needed that tooling. I'm moderately afraid of the tooling, and it doesn't handle renaming directories (just files). i.e. I need to improve it a lot. (It also interferes with how the rest of my tooling works, since I only try to change files that I haven't excluded, but a renamed file would no longer be on the list of files to touch, and thus...)

**Comment by jsoref** [comment]
Sorry about all the pushes. I have a repo where I do my work, but I turned off the workflows because they were starving my workflow, and I forgot to turn them back on.

File: fonts/bravura/FONTLOG.txt
Comment (jsoref) [reply fonts/bravura/FONTLOG.txt:172]: https://github.com/steinbergmedia/petaluma/pull/13

Comment (jsoref) [reply fonts/bravura/FONTLOG.txt:172]: Did you mean https://github.com/steinbergmedia/bravura?

Comment (jsoref) [reply fonts/bravura/FONTLOG.txt:172]: Oh, nope, this is another from smufl. PR for that shortly.

File: src/importexport/guitarpro/internal/gtp/gp67dombuilder.cpp
Comment (jsoref) [inline src/importexport/guitarpro/internal/gtp/gp67dombuilder.cpp:56]: Oh, I should call this out. This random typo is was an unused variable. Fixing the typo of course broke the build since it tripped over the other `rhythms` object which has a totally different shape.

File: src/engraving/libmscore/layoutbreak.cpp
Comment (jsoref) [inline src/engraving/libmscore/layoutbreak.cpp:115]: This is the API change -- I've split it into its own commit so it could be addressed separately.

File: src/importexport/guitarpro/internal/gtp/gpconverter.cpp
Comment (jsoref) [inline src/importexport/guitarpro/internal/gtp/gpconverter.cpp:266]: Sigh, this isn't a proper correction. It's a hazard of substrings and sorting, I need to change my tooling to sort longer matches before shorter matches -- but right now I use Google Sheets and just let it do sorting, but it clearly favors `denom` over `denominator`.

File: fonts/smufl/glyphnames.json
Comment (jsoref) [reply fonts/smufl/glyphnames.json:1805]: Filed https://github.com/w3c/smufl/pull/237

File: fonts/bravura/FONTLOG.txt
Comment (jsoref) [reply fonts/bravura/FONTLOG.txt]: Moved to https://github.com/jsoref/MuseScore/commits/spelling-later while it waits for the smufl PR to resolve.

File: fonts/smufl/glyphnames.json
Comment (jsoref) [reply fonts/smufl/glyphnames.json]: https://github.com/musescore/MuseScore/pull/11566#discussion_r867539770

File: fonts/petaluma/OFL-FAQ.txt
Comment (jsoref) [reply fonts/petaluma/OFL-FAQ.txt]: https://github.com/musescore/MuseScore/pull/11566#discussion_r867539770

File: fonts/bravura/FONTLOG.txt
Comment (jsoref) [reply fonts/bravura/FONTLOG.txt]: https://github.com/musescore/MuseScore/pull/11566#discussion_r867539770

**Comment by cbjeukendrup** [comment]
Rebase needed!

**Review by cbjeukendrup — COMMENTED** [review]
What a lot of work! Thanks very much. I have gone through it, and left a few comments.

(By the way, I saw you did the same thing at the Swift project, one of my favorite programming languages 😉)

File: mscore3.txt
Comment (jsoref) [reply mscore3.txt:216]: Apparently. I can never remember this one. I should have searched.

File: src/importexport/musicxml/tests/tst_mxml_io.cpp
Comment (cbjeukendrup) [inline src/importexport/musicxml/tests/tst_mxml_io.cpp:156]: Oops, you forgot to remove ` fails` at the end of the line

File: src/engraving/libmscore/splitMeasure.cpp
Comment (cbjeukendrup) [inline src/engraving/libmscore/splitMeasure.cpp:96]: Sorry, still one word missing here: 
...the beginning **_of_** the split measure...

File: src/engraving/compat/midi/midiinstrument.cpp
Comment (cbjeukendrup) [inline src/engraving/compat/midi/midiinstrument.cpp:427]: I really don't feel confident about this. Would you mind dropping the three other changes to this file too?

**Comment by cbjeukendrup** [comment]
@jsoref (Probably) the final few comments:
- It would be great if you could fix this one: https://github.com/musescore/MuseScore/pull/11566#discussion_r867435610 (about `read` instead of `readed` in `ipc.{h,cpp}`)
- I'd vote to drop all `dstacattissimo`/`dstaccatissimo` changes, since I'm not sure whether it's good to change that, and that whole `test` folder is obsolete anyway
- You'll need to rebase to fix conflicts once more.

Thanks!

**Comment by jsoref** [comment]
@cbjeukendrup: afaict the default branch is broken... You should be able to push to this branch if you want to. I'm done for the weekend.

**Comment by Jojo-Schmitz** [comment]
Right:
```
...\MuseScore\src\framework\audio\internal\audioconfiguration.h:26: Fehler: system/ifilesystem.h: No such file or directory
In file included from C:/Users/Jojo/Documents/GitHub/MuseScore/src/framework/audio/audiomodule.cpp:30,
                 from ...\MuseScore\build.qtc\src\framework\audio\CMakeFiles\audio.dir\Unity\unity_0_cxx.cxx:5:
.../MuseScore/src/framework/audio/internal/audioconfiguration.h:26:10: fatal error: system/ifilesystem.h: No such file or directory
 #include "system/ifilesystem.h"
          ^~~~~~~~~~~~~~~~~~~~~~
```

**Comment by cbjeukendrup** [comment]
See #11629 :)

@jsoref Thanks again for your work! If necessary, I will make some final fixes myself, and then merge it.

**Comment by cbjeukendrup** [comment]
@jsoref It looks like you accidentally undid some changes of yours with the last force-push. But no worries, I already fixed it on my local branch. I would like to push the changes to this PR, so that I can merge it, but that is not working. Did you by chance uncheck the checkbox called "Allow edits and access to secrets by maintainers", and if so, would you mind enabling it (temporarily)?

(I would prefer to push the changes to this PR first, before merging them to master, so that we can easily see here in the future what exactly has been merged.)

**Comment by jsoref** [comment]
image

**Comment by jsoref** [comment]
Sorry, I have two computers and it's quite possible I was using the wrong one...

**Review by cbjeukendrup — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
@jsoref Everything should be fine now. Thanks for your work!

**Pull request merged**
